cocos/2d/CCMotionStreak.cpp

```diff
@@ -126,7 +127,7 @@ bool MotionStreak::initWithFade(float fade, float minSeg, float stroke, const Co
     _blendFunc = BlendFunc::ALPHA_NON_PREMULTIPLIED;
 
     // shader state
-    setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR));
+    setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_A8_COLOR));
```

Comment:
@minggo 

It has been over a year since I filed this pull request. I don't use cocos2d-x at the moment nor do I even have the code at my computer. In short, I can't help you further.

If I remember correctly there was some issue with the shader, even though I never fully understood that issue. 
The easiest solution is to keep the current MotionStreak class, and write some documentation on how to change it to add opacity support, or derive some MotionStreakOpacity class from it which provides opacity support, perhaps at the cost of some other feature.